### PR TITLE
Travis spellcheck and cppunit with valgrind

### PR DESCRIPTION
**Pull request by jimklimov**
For issue #338 

This enables Travis tests for cppunit (to whatever extent was available in codebase, which was not much), and fixes spellchecking to do its work (and also fixes a number of discovered spelling errors).

Also adds configure capability to find and optionally enable valgrind, but currently does not go beyond that (e.g. enable debug builds and/or run any special checks under valgrind... not sure how/what to test here - start every binary to display help text and see that nothing went wrong? the commented-away code for travis enablement was slightly improved but is still commented away). This piece is ported from DMF branch which does have a target to make a custom build of everything it needed to test that the parser does not leak.

UPDATE: For lack of better ideas at the moment, added a distcheck taryet with valgrind support that runs the cppunit test programs with leak inspection. This approach can be replicated to add other types of 'make check' as they come.

### CODE REVIEW

**Comment by jimklimov** [comment]
Ok, so after some debugging to drop later - the mismatch is in aspell shipped dictionaries on travis hosts == more exceptions needed in nut.dict.

**Comment by aquette** [comment]
Cool, so that worked with modifying the first line of nut.dict with utf8 ;)
Thx for all your work on NUT @jimklimov
Happy to finally see the actual use of all these personal nights of
works!!!!
I owe you some beers ... or vodka ;)

**Comment by jimklimov** [comment]
I did not yet experiment with utf8 header nor other options, so contributor
names are still a problem :(

--
Typos courtesy of GMail on my Samsung Android

19.01.2017 23:11 пользователь "Arnaud Quette" <notifications@github.com>
написал:

**Comment by jimklimov** [comment]
Seems finally fixed for "UTF8" input and dicts (just one Pawel with polish L got missed by aspell and is now seen as Pawe with a skipped next character that is otherwise deemed invalid in a word), so un-WIPing the PR.

**Comment by jimklimov** [comment]
@aquette : PS: I am not quite a vodka guy, but I get the drift ;)
Kudos to you too for getting a solid foundation (and hints) without which I wouldn't get so far with the spells.

File: docs/Makefile.am [hunk 1]
```diff
@@ -125,17 +125,75 @@ SPELLCHECK_SRC = $(ALL_TXT_SRC) ../README ../INSTALL.nut ../UPGRADING  ../NEWS \
...
+# international committers). The grep tests below look if the aspell output
+# contained something other than the OK lines (tagged with asterisk) and
+# aspell's version (tagged with @) and if it did - those lines must be the
+# spellcheck complaints. Empty OUT is ok.
+# We also must indent the input, because certain piped-in characters are
+# interpreted as commands, and seems this feature can not be turned off.
+# See also http://aspell.net/man-html/Through-A-Pipe.html
+# TODO: Is "grep -a" or "grep -b" (treat input as ascii/bin) portable enough?
+# Is "egrep == grep -E" always valid? (maybe all a job for configure.ac)
+# TODO: Define a custom language in-repo or during build, that is english
```
Comment (aquette) [inline docs/Makefile.am]: is this and the below still valid?

File: docs/Makefile.am [hunk 2]
```diff
@@ -125,17 +125,75 @@ SPELLCHECK_SRC = $(ALL_TXT_SRC) ../README ../INSTALL.nut ../UPGRADING  ../NEWS \
 	../TODO ../scripts/ufw/README ../scripts/augeas/README ../lib/README \
 	../tools/nut-scanner/README
```
Comment (aquette) [inline docs/Makefile.am]: add daisychain.txt to ALL_TXT_SRC since it's missing

**Comment by aquette** [comment]
@jimklimov great work! and thx for the thank ;)
on the valgrind, we can improve that when integrating QRT and use it (similar approach, with dummy-ups+upsd+upsmon+upsc/rw) to check for leaks on these. Nearly impossible for the drivers... at least for now.

File: docs/Makefile.am (hunk 1 above)
Comment (jimklimov) [inline docs/Makefile.am]: Custom language - probably not anymore, at least not until we hit a platform with sufficiently different or old aspell.

Concerns of portable `egrep` and `-a/-b` flags - these may still be valid (buildbots might show what works where). If the issues do pop up, the flag may be removed at expense of warnings that unexpected binary input was detected - and those might need to be grepped out so as to not pollute logged reports and/or interpretation of aspell's "$OUT".

File: docs/Makefile.am (hunk 2 above)
Comment (jimklimov) [inline docs/Makefile.am]: done

**Comment by jimklimov** [comment]
Good point on manpages - done. No copy-paste involved ;)

**Comment by jimklimov** [comment]
For reviewers : so far my fantasy is exhausted as far as "small initial steps" go, and this PR looks solid enough for merging, as a foundation for possible (or inevitable) improvements. I'd like to avoid it growing into a monster like DMF PRs that take months to review, and to begin bringing benefits to subsequent codebase improvements ASAP ;)

**Comment by aquette** [comment]
@jimklimov thx, you've addressed all the points from my review. Good to go, and as you said, this is already improving the situation, along with being a solid foundation to build upon.
Now merging / closing the PR

**Pull request merged**